Updating Github Actions workflow to use Virtual Satellite actions

PhilMFischer opened the pull request:
- Updated workflow to use Virrtual Satellite Actions in version v2.
- Outsourced all scripts under bash folder into Virtual Satellite actions.

Closes #134

Commit:
Split maven tycho build for spotbugs, checkstyle, surefire - (Task #134)

- Added three new profiles for spotbugs, checkstyle and surefire, to
have a similar build process as in Virtula Satellite. This will allow
reusing Virtual Satellite actions for GitHub actions.
- Added launchers for new profile
- Split build also into product and deploy

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Removed Travis-CI workflow and bash script - (Task #134)

- Removed travis ci since it will be replaced by GitHUb action workflow
- Removed bash script for upload since it will be replaced by common
scripts of VirtualSatellite GitHub actions

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Added new GitHub Workflow file - (Task #134)



---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Removed verify dependency from github workflow - (Task #134)

- verify job was still mentioned in other jobs. verify is not used in
Overtarget since we do not check thee authors.

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Fixed case of image file name extension - (Task #134)

- AsciiDoc build was not correct due to incorrect case of image file
name xtension

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Added new encyrpted SF secret to repository - (Task #134)

- Added new encrypted ssh key to repository for SF deployments

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Activated deploy for testing - (Task #134)


---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Updated password of ssh key encrption - (Task #134)

- Fixed issue with ssh openssl password

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Fixing SF project name - (Task #134)

- using overtarget with lowercase o instead of capital O

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Updated build to deploy zipped p2s as binary - (Task #134)

- Also updated launcher.

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Updated build to deploy zipped p2s as binary - (Task #134)

- Also updated launcher.

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Blocking deploy for non releasing builds - (Task #134)

---
Task #134: Migrate from Travis-CI to GitHub Actions

Commit:
Using v2 version of virtual satellite actions - (Task #134)

- Changed workflow to use virtual satellite actions in version v2

---
Task #134: Migrate from Travis-CI to GitHub Actions

Anna-Lena-DLR reviewed (approved):
Looks good 👍

Commit:
Update Url in Readme - (Task #134)

- Update badge Urls to point to GitHUb actions

---
Task #134: Migrate from Travis-CI to GitHub Actions

franzTobiasDLR commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,289 @@
+on:
+  # --------------------------------------------
+  # Controls when the automatic build job will be run.
+  # By default uses the default branch. Runs at 03:00 UTC.
```
Outdated comment? It runs at 05:00 UTC, right? Maybe we could remove that sentence completely...

Commit:
Fixing evaluation of GITHUB_SHA environment variable - (Task #134)

- environment variable is case sensitive

---
Task #134: Migrate from Travis-CI to GitHub Actions
